components/support/utils/src/main/java/mozilla/components/support/utils/Performance.kt

```diff
@@ -0,0 +1,25 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.support.utils
+
+import mozilla.components.support.base.log.logger.Logger
+
+/**
+ * Executes the given [block] and logs the elapsed time in milliseconds.
+ * Uses [System.nanoTime] for measurements, since it isn't tied to a wall-clock.
+ *
+ * @param logger [Logger] to use for logging.
+ * @param op Name of the operation [block] performs.
+ * @param block A lambda to measure.
+ * @return [T] result of running [block].
+ */
+@SuppressWarnings("MagicNumber")
+inline fun <T> logElapsedTime(logger: Logger, op: String, block: () -> T): T {
```

Comment:
Neat! I think I've seen code like this before, maybe @mcomella had something like it written in Fenix? I can't seem to find it now though.

Reply:
Didn't see anything like what I wanted; we can start cleaning up similar code now.